obspy/io/reftek/core.py

```diff
@@ -0,0 +1,341 @@
+                "Time",
+                "  | Packet Type   |  Event #  | Station | Channel #         "
+                "|",
+                "  |   |  Unit ID  |    | Data Stream #  |   |  # of samples "
+                "|",
+                "  |   |   |  Exper.#   |   |  |  |      |   |    |          "
+                "|"]
+            for data in self._data:
+                info.append(Packet.from_data(data).__str__(compact=True))
+            info.append("(detailed packet information with: "
+                        "'print(Reftek130.__str__(compact=False))')")
+        else:
+            info = ["Reftek130 ({:d} packets)".format(len(self._data))]
+            for data in self._data:
+                info.append(str(Packet.from_data(data)))
+        return "\n".join(info)
+
+    @staticmethod
+    def from_file(filename):
+        with io.open(filename, "rb") as fh:
+            string = fh.read()
+        rt = Reftek130()
+        rt._data = _initial_unpack_packets(string)
+        return rt
+
+    def check_packet_sequence_and_sort(self):
+        """
+        Checks if packet sequence is ordered. If not, shows a warning and sorts
+        packets by packet sequence. This should ensure that data (DT) packets
+        are properly enclosed by the appropriate event header/trailer (EH/ET)
+        packets.
+        """
+        if np.any(np.diff(self._data['packet_sequence']) - 1 < 0):
+            msg = ("Detected permuted packet sequence, sorting.")
+            warnings.warn(msg)
+            self._data.sort(order=native_str("packet_sequence"))
+
+    def check_packet_sequence_contiguous(self):
+        """
+        Checks if packet sequence is contiguous, i.e. without missing packets
+        in between. Currently raises if that is the case because this case is
+        not covered by test data yet.
+        """
+        if np.any(np.diff(self._data['packet_sequence']) - 1 != 0):
+            msg = ("Detected a non-contiguous packet sequence!")
+            warnings.warn(msg)
+
+    def drop_not_implemented_packet_types(self):
+        """
+        Checks if there are packets of a type that is currently not implemented
+        and drop them showing a warning message.
+        """
+        is_implemented = np.in1d(
+            self._data['packet_type'],
+            [x.encode() for x in PACKET_TYPES_IMPLEMENTED])
+        if not np.all(is_implemented):
+            not_implemented = np.invert(is_implemented)
+            count_not_implemented = not_implemented.sum()
+            types_not_implemented = np.unique(
+                self._data['packet_type'][not_implemented])
+            msg = ("Encountered some packets of types that are not "
+                   "implemented yet (types: {}). Dropped {:d} packets "
+                   "overall.")
+            msg = msg.format(types_not_implemented.tolist(),
+                             count_not_implemented)
+            warnings.warn(msg)
+            return
```

Comment:
Euuuh.. nice catch, don't know how this could get so garbled up, must've gone wrong during squashing and rebasing..